spec.md

```diff
@@ -2888,3 +2888,32 @@ services:
   foo:
     build: null
 ```
+## Dependency on other compose projects
+
+A Compose project can declare dependency on another compose project, managed by another team
+or shared with others. This allows to keep a compse file reasonably complicated for the limited
+amount of resources a single team has to declare for it's own sub-domain within a larger deployment.
+
+`require` top-level section is used to define dependency to another compose application (or subdomain).
+Each path listed in this section MUST be loaded as an individual compose model with it's own project
+directory to resolve relative paths. The resulting resources can be used in the including compose model
+for cross-services references.
+
+```yaml
+require:
+  - ../commons/compose.yaml
+  - ../another_domain/compose.yaml
+
+services:
+  webapp:
+    depends_on:
+      - imported-service # defined by another_domain
```

Comment:
i'm a little bit worried about automatically importing all of "another_domain"s names into the main project's name space. When I was prototyping this with real projects, I ran into this problem very quickly:

User Service Compose File:
```
services:
  api: ...
  db: ...
```

Repo Service Compose File
```
services:
  api: ...
  db: ...
requires:
- ./users.compose.yaml
```

where you really don't want to merge the user DB and repo DB together, you want them to be separate DBs. 

I feel like we're going to need something more like:

```
imports:
  - from: ./users.compose.yaml
    service: api 
    as: users-api
```

where both the users project and repos project both get loaded, but they have their own networks and we have ways to link them without overwriting names.

Reply:
if you only need a single service, especially if you need to rename or customize, better use `extends` (which was the conclusion on https://github.com/docker/compose/issues/318)

Reply:
`extends` is designed for this use case:

```yaml
services:
  users-api:
    extends: 
      file : ./users.compose.yaml
      service: api 
```

Reply:
hmm...just tried that and it didn't behave how i'm describing, so i think there's a misunderstanding.

in my use case, i want to pull in the whole Users project.

users.compose.yaml is maintained by another team. I don't know what services they need to run. They should be able to add services to their compose file without breaking the main project.

In your example, extends only pulls in the api service, not the other services in the file. changes to the users.compose.yaml file would break all the extending projects.

Reply:
right, then this is the exact goal for this feature. And corner cases like includes/excludes or renaming can be implemented as a second interation